judger/src/tester/model.rs

```diff
@@ -37,10 +35,11 @@ impl Bind {
     }
 }
 
-pub fn path_canonical_from(path: &Path, base_dir: &Path) -> PathBuf {
-    let mut from_base = base_dir.to_owned();
-    from_base.push(path);
-    from_base.absolutize().unwrap().into_owned()
+/// Push the `relative` path to `base` path and canonicalize the result.
+pub fn canonical_push(base: &Path, relative: &Path) -> PathBuf {
```

Comment:
`canonical_join` should be better than `..._push`, since the definition in stdlib is `Path::join(&Path, &Path) -> PathBuf` and `PathBuf::push(&mut self, Into<Path>) -> ()`

Reply:
Resolved in a76c30b.